highgui qt compile error multiple definition of

opencv master 29b0ef44aa24e237498b032c2ab5adc63ffafadb  
g++ (GCC) 7.2.1 20170915 (Red Hat 7.2.1-2)
Using Qt version 5.7.1 in /usr/lib64

`CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvButtonbar::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN11CvButtonbar18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvButtonbar::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN11CvButtonbar18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `GuiReceiver::metaObject() const':
mocs_compilation.cpp:(.text._ZNK11GuiReceiver10metaObjectEv+0x0): multiple definition of `GuiReceiver::metaObject() const'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZNK11GuiReceiver10metaObjectEv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o:(.data.rel.ro+0x200): multiple definition of `GuiReceiver::staticMetaObject'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:(.data.rel.ro+0x200): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvButtonbar::metaObject() const':
mocs_compilation.cpp:(.text._ZNK11CvButtonbar10metaObjectEv+0x0): multiple definition of `CvButtonbar::metaObject() const'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZNK11CvButtonbar10metaObjectEv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o:(.data.rel.ro+0x1c0): multiple definition of `CvButtonbar::staticMetaObject'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:(.data.rel.ro+0x1c0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvPushButton::metaObject() const':
mocs_compilation.cpp:(.text._ZNK12CvPushButton10metaObjectEv+0x0): multiple definition of `CvPushButton::metaObject() const'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZNK12CvPushButton10metaObjectEv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o:(.data.rel.ro+0x180): multiple definition of `CvPushButton::staticMetaObject'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:(.data.rel.ro+0x180): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvCheckBox::metaObject() const':
mocs_compilation.cpp:(.text._ZNK10CvCheckBox10metaObjectEv+0x0): multiple definition of `CvCheckBox::metaObject() const'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZNK10CvCheckBox10metaObjectEv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o:(.data.rel.ro+0x140): multiple definition of `CvCheckBox::staticMetaObject'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:(.data.rel.ro+0x140): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvRadioButton::metaObject() const':
mocs_compilation.cpp:(.text._ZNK13CvRadioButton10metaObjectEv+0x0): multiple definition of `CvRadioButton::metaObject() const'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZNK13CvRadioButton10metaObjectEv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o:(.data.rel.ro+0x100): multiple definition of `CvRadioButton::staticMetaObject'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:(.data.rel.ro+0x100): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvTrackbar::metaObject() const':
mocs_compilation.cpp:(.text._ZNK10CvTrackbar10metaObjectEv+0x0): multiple definition of `CvTrackbar::metaObject() const'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZNK10CvTrackbar10metaObjectEv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o:(.data.rel.ro+0xc0): multiple definition of `CvTrackbar::staticMetaObject'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:(.data.rel.ro+0xc0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvWinProperties::metaObject() const':
mocs_compilation.cpp:(.text._ZNK15CvWinProperties10metaObjectEv+0x0): multiple definition of `CvWinProperties::metaObject() const'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZNK15CvWinProperties10metaObjectEv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o:(.data.rel.ro+0x80): multiple definition of `CvWinProperties::staticMetaObject'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:(.data.rel.ro+0x80): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvWindow::metaObject() const':
mocs_compilation.cpp:(.text._ZNK8CvWindow10metaObjectEv+0x0): multiple definition of `CvWindow::metaObject() const'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZNK8CvWindow10metaObjectEv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o:(.data.rel.ro+0x40): multiple definition of `CvWindow::staticMetaObject'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:(.data.rel.ro+0x40): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `DefaultViewPort::metaObject() const':
mocs_compilation.cpp:(.text._ZNK15DefaultViewPort10metaObjectEv+0x0): multiple definition of `DefaultViewPort::metaObject() const'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZNK15DefaultViewPort10metaObjectEv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o:(.data.rel.ro+0x0): multiple definition of `DefaultViewPort::staticMetaObject'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:(.data.rel.ro+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvButtonbar::qt_metacast(char const*)':
mocs_compilation.cpp:(.text._ZN11CvButtonbar11qt_metacastEPKc+0x0): multiple definition of `CvButtonbar::qt_metacast(char const*)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN11CvButtonbar11qt_metacastEPKc+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvTrackbar::qt_metacast(char const*)':
mocs_compilation.cpp:(.text._ZN10CvTrackbar11qt_metacastEPKc+0x0): multiple definition of `CvTrackbar::qt_metacast(char const*)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN10CvTrackbar11qt_metacastEPKc+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvButtonbar::qt_metacall(QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN11CvButtonbar11qt_metacallEN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvButtonbar::qt_metacall(QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN11CvButtonbar11qt_metacallEN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvTrackbar::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN10CvTrackbar18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvTrackbar::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN10CvTrackbar18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvTrackbar::qt_metacall(QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN10CvTrackbar11qt_metacallEN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvTrackbar::qt_metacall(QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN10CvTrackbar11qt_metacallEN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvWindow::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN8CvWindow18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvWindow::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN8CvWindow18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `DefaultViewPort::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN15DefaultViewPort18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): multiple definition of `DefaultViewPort::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN15DefaultViewPort18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvWinProperties::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN15CvWinProperties18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvWinProperties::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN15CvWinProperties18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvPushButton::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN12CvPushButton18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvPushButton::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN12CvPushButton18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvRadioButton::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN13CvRadioButton18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvRadioButton::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN13CvRadioButton18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvCheckBox::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN10CvCheckBox18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvCheckBox::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN10CvCheckBox18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `GuiReceiver::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN11GuiReceiver18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): multiple definition of `GuiReceiver::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN11GuiReceiver18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `GuiReceiver::qt_metacast(char const*)':
mocs_compilation.cpp:(.text._ZN11GuiReceiver11qt_metacastEPKc+0x0): multiple definition of `GuiReceiver::qt_metacast(char const*)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN11GuiReceiver11qt_metacastEPKc+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `GuiReceiver::qt_metacall(QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN11GuiReceiver11qt_metacallEN11QMetaObject4CallEiPPv+0x0): multiple definition of `GuiReceiver::qt_metacall(QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN11GuiReceiver11qt_metacallEN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvPushButton::qt_metacast(char const*)':
mocs_compilation.cpp:(.text._ZN12CvPushButton11qt_metacastEPKc+0x0): multiple definition of `CvPushButton::qt_metacast(char const*)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN12CvPushButton11qt_metacastEPKc+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvPushButton::qt_metacall(QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN12CvPushButton11qt_metacallEN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvPushButton::qt_metacall(QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN12CvPushButton11qt_metacallEN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvCheckBox::qt_metacast(char const*)':
mocs_compilation.cpp:(.text._ZN10CvCheckBox11qt_metacastEPKc+0x0): multiple definition of `CvCheckBox::qt_metacast(char const*)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN10CvCheckBox11qt_metacastEPKc+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvCheckBox::qt_metacall(QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN10CvCheckBox11qt_metacallEN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvCheckBox::qt_metacall(QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN10CvCheckBox11qt_metacallEN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvRadioButton::qt_metacast(char const*)':
mocs_compilation.cpp:(.text._ZN13CvRadioButton11qt_metacastEPKc+0x0): multiple definition of `CvRadioButton::qt_metacast(char const*)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN13CvRadioButton11qt_metacastEPKc+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvRadioButton::qt_metacall(QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN13CvRadioButton11qt_metacallEN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvRadioButton::qt_metacall(QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN13CvRadioButton11qt_metacallEN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvWinProperties::qt_metacast(char const*)':
mocs_compilation.cpp:(.text._ZN15CvWinProperties11qt_metacastEPKc+0x0): multiple definition of `CvWinProperties::qt_metacast(char const*)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN15CvWinProperties11qt_metacastEPKc+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvWindow::qt_metacast(char const*)':
mocs_compilation.cpp:(.text._ZN8CvWindow11qt_metacastEPKc+0x0): multiple definition of `CvWindow::qt_metacast(char const*)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN8CvWindow11qt_metacastEPKc+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvWinProperties::qt_metacall(QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN15CvWinProperties11qt_metacallEN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvWinProperties::qt_metacall(QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN15CvWinProperties11qt_metacallEN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvWindow::qt_metacall(QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN8CvWindow11qt_metacallEN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvWindow::qt_metacall(QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN8CvWindow11qt_metacallEN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `DefaultViewPort::qt_metacast(char const*)':
mocs_compilation.cpp:(.text._ZN15DefaultViewPort11qt_metacastEPKc+0x0): multiple definition of `DefaultViewPort::qt_metacast(char const*)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN15DefaultViewPort11qt_metacastEPKc+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `DefaultViewPort::qt_metacall(QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN15DefaultViewPort11qt_metacallEN11QMetaObject4CallEiPPv+0x0): multiple definition of `DefaultViewPort::qt_metacall(QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN15DefaultViewPort11qt_metacallEN11QMetaObject4CallEiPPv+0x0): first defined here
collect2: error: ld returned 1 exit status
make[2]: *** [modules/highgui/CMakeFiles/opencv_highgui.dir/build.make:257: lib/libopencv_highgui.so.3.3.0] Error 1
make[1]: *** [CMakeFiles/Makefile2:5408: modules/highgui/CMakeFiles/opencv_highgui.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
`

--- a/modules/highgui/CMakeLists.txt
+++ b/modules/highgui/CMakeLists.txt
@@ -38,7 +38,9 @@ file(GLOB highgui_ext_hdrs
 list(REMOVE_ITEM highgui_ext_hdrs "${CMAKE_CURRENT_LIST_DIR}/include/opencv2/${name}/highgui_winrt.hpp")
 
 if(HAVE_QT5)
-  set(CMAKE_AUTOMOC ON)
+  # "Automoc" doesn't work properly with opencv_world build, use QT5_WRAP_CPP() directly
+  #set(CMAKE_AUTOMOC ON)
+
   set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
   QT5_ADD_RESOURCES(_RCC_OUTFILES ${CMAKE_CURRENT_LIST_DIR}/src/window_QT.qrc)

